### PR TITLE
Add search and friendly name to integration table

### PR DESCRIPTION
**Pull request by ludeeus**
- Concat integration list from the doc site to indicate "unused" integrations.
- Use the title from the doc site if any
- Add filter and index to integration list

### CODE REVIEW

File: site/src/data.ts [hunk 1]
```diff
@@ -27,9 +27,21 @@ export interface CurrentAnalytics extends Analytics {
...
+export interface IntegrationData {
+  integration: string;
+  installations: number;
+}
+
 export const fetchData = () =>
   fetch("https://analytics-api.home-assistant.io/v1");
 
+export const fetchIntegrationDetails = () =>
+  fetch("https://www.home-assistant.io/integrations.json", { mode: "no-cors" });
```
Comment (bramkragten) [inline site/src/data.ts:43]: Isn't that already in `window.integrationsPromise`?

File: site/src/analytics-integrations.ts [hunk 2]
```diff
@@ -124,6 +177,50 @@ export class AnalyticsIntegrations extends LitElement {
...
+
+  private _clearFilter() {
+    this._currentTablePage = 0;
+    this._filter = "";
+  }
+
+  async getData() {
+    try {
+      const response = await ((window as any).integrationsPromise ||
+        fetchIntegrationDetails());
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:199]: What could cause that `integrationsPromise` is not defined?

File: site/src/data.ts (hunk 1 above)
Comment (ludeeus) [reply site/src/data.ts:43]: It is

File: site/src/analytics-integrations.ts (hunk 2 above)
Comment (ludeeus) [reply site/src/analytics-integrations.ts:199]: Time, the same approach is currently used for the worker data

File: site/src/analytics-integrations.ts [hunk 3]
```diff
@@ -124,6 +177,50 @@ export class AnalyticsIntegrations extends LitElement {
...
+      if (response.ok) {
+        this._integrationDetails = await response.json();
+
+        this._integrations = this._integrations!.concat(
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:203]: We could save some work by doing the `_computeName` here

File: site/src/analytics-integrations.ts [hunk 4]
```diff
@@ -12,77 +11,114 @@ import {
...
+      </div>
+
       <table>
         <tr class="table-header">
+          ${!isMobile ? html`<th></th>` : ""}
           <th>Integration</th>
           <th>Installations</th>
         </tr>
         ${tableData.map(
           (entry) => html`
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:108]:
```suggestion
          (entry, idx) => html`
```

File: site/src/analytics-integrations.ts [hunk 5]
```diff
@@ -12,77 +11,114 @@ import {
...
         (a, b) =>
-          b.installations - a.installations ||
-          a.integration.localeCompare(b.integration)
+          b.installations - a.installations || a.title.localeCompare(b.title)
       )
-      .slice(tableStart, tableEnd);
+      .map((entry) => {
+        idx++;
+        return { ...entry, idx };
+      })
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:60]:
```suggestion
```

File: site/src/analytics-integrations.ts [hunk 6]
```diff
@@ -12,77 +11,114 @@ import {
...
     if (this._integrations === undefined) {
       return html``;
     }
 
-    const tableStart = this._currentTablePage * this._currentTableSize;
-    const tableEnd =
-      tableStart + this._currentTableSize <= this._integrations.length
-        ? tableStart + this._currentTableSize
-        : this._integrations.length;
+    let idx = 0;
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:50]:
```suggestion
```

File: site/src/analytics-integrations.ts [hunk 7]
```diff
@@ -124,6 +160,57 @@ export class AnalyticsIntegrations extends LitElement {
...
+    this._filter = "";
+  }
+
+  async getData() {
+    const dataKeys = Object.keys(this.data!);
+    const lastEntry = this.data![dataKeys[dataKeys.length - 1]];
+    try {
+      const response = await ((window as any).integrationsPromise ||
+        fetchIntegrationDetails());
+      if (response.ok) {
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:179]:
```suggestion
      if (!response.ok) {
      	return;
      }
```

File: site/src/analytics-integrations.ts [hunk 8]
```diff
@@ -124,6 +160,57 @@ export class AnalyticsIntegrations extends LitElement {
...
+        this._integrations = Object.keys(lastEntry.integrations)
+          .map((domain) => {
+            return {
+              domain,
+              title: this._integrationDetails[domain]?.title || domain,
+              installations: lastEntry.integrations[domain] || 0,
+            };
+          })
+          .concat(
+            Object.keys(this._integrationDetails)
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:191]: Can't we walk over this list as this has all integrations and then check if it is in the data?

File: site/src/analytics-integrations.ts (hunk 4 above)
Comment (ludeeus) [reply site/src/analytics-integrations.ts:105]: Did that in L55 to keep placement when using search

Comment (bramkragten) [reply site/src/analytics-integrations.ts:105]: Ah ok :-)

File: site/src/analytics-integrations.ts [hunk 9]
```diff
@@ -124,6 +157,46 @@ export class AnalyticsIntegrations extends LitElement {
...
+      if (!response.ok) {
+        return;
+      }
+
+      this._integrationDetails = await response.json();
+
+      Object.keys(lastEntry.integrations)
+        .concat(Object.keys(this._integrationDetails))
+        .forEach((entry) => domains.add(entry));
+      console.log(domains);
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:186]: Console!

File: site/src/analytics-integrations.ts [hunk 10]
```diff
@@ -124,6 +157,46 @@ export class AnalyticsIntegrations extends LitElement {
...
+        fetchIntegrationDetails());
+      if (!response.ok) {
+        return;
+      }
+
+      this._integrationDetails = await response.json();
+
+      Object.keys(lastEntry.integrations)
+        .concat(Object.keys(this._integrationDetails))
+        .forEach((entry) => domains.add(entry));
```
Comment (bramkragten) [inline site/src/analytics-integrations.ts:185]:
```ts
domain = new Set(Object.keys(lastEntry.integrations)
        .concat(Object.keys(this._integrationDetails)))
        ```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**